infra/modules/providers/azure/service-plan/main.tf

```diff
@@ -0,0 +1,27 @@
+resource "random_id" "remotestate_account_name" {
+  byte_length = 6
+  keepers {
+    sa_account_ref = 1
+  }
+}
+
+resource "azurerm_resource_group" "svcplan" {
+  name     = "${var.resource_group_name == "" ? "${local.name}-cobalt-rg" : "${var.resource_group_name}"}"
+  location = "${var.resource_group_location}"
+  tags     = "${merge(map("Name", "${local.name}"), var.resource_tags)}"
+}
+
+resource "azurerm_app_service_plan" "svcplan" {
+  name                = "${var.svcplan_name == "" ? "${local.name}-cobalt-svcplan" : "${var.svcplan_name}"}"
```

Comment:
The `name` should be driven solely on the `svcplan_name` var

Reply:
It is a 3 step process and I agree on the way bedrock is doing it: 
1. Check if `svcplan_name` is provided, if provided assign it to name. 
2. Else check if `local.name` is provided by the user to be added as a prefix, 
3. if not provided then it will generate a randomized `local.name` prefix for the name.

Reply:
This approach feels too opinionated for cobalt clients. This is a departure from the approach that bedrock [takes ](https://github.com/Microsoft/bedrock/blob/master/cluster/azure/aks/main.tf#L11)which contains no logic on aks name provisioning. I'd prefer to remove `local.name` all together and just `service_plan_name`.